modules/l4proxy/matcher.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 Matthew Holt
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4proxy
+
+import (
+	"bufio"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/mastercactapus/proxyprotocol"
+	"github.com/mholt/caddy-l4/layer4"
+)
+
+func init() {
+	caddy.RegisterModule(MatchPROXY{})
+}
+
+type MatchPROXY struct{}
```

Comment:
Do we want the ability to match against the source IP/Port? Right now, it matches a connection, as long as it finds a PROXY protocol header.

Reply:
That would be a separate matcher that already exists:

https://github.com/mholt/caddy-l4/blob/3cfcafe708834b071dae82a5f75224e55baef230/layer4/matchers.go#L95-L100

Reply:
Sure, I did see that matcher, but if Caddy is behind another proxy, then this would match against the proxy's IP rather than the original connection's IP which is available in the PROXY header. 

As I write this comment though, it occurs to me that it might not be a valid use-case to expect Caddy (while placed behind another proxy) to match against original connection's IP addresses.

Reply:
One of the reasons why I used that library initially, since it has the logic to parse the PROXY header fully and retrieve source/destination ip address/port.

Reply:
Ah, well, let's wait and see if an actual use case appears first.

Reply:
Oh actually, can we rename this to `MatchProxyProtocol` first?

Reply:
Have renamed it.